fastlane/lib/fastlane/actions/testfairy.rb

```diff
@@ -17,8 +17,58 @@ def self.run(params)
 
         return params[:ipa] if Helper.test?
 
-        client_options = params.values
-        client_options[:testers_groups] = params[:testers_groups].join(',')
+        metrics_to_client = lambda do |metrics|
+          metrics.map do |metric|
+            case metric
+            when :cpu, :memory, :network, :gps, :battery, :mic, :wifi
+              metric.to_s
+            when :phone_signal
+              'phone-signal'
+            else
+              UI.user_error!("Unknown metric: #{metric}")
+            end
+          end
+        end
+
+        options_to_client = lambda do |options|
+          options.map do |option|
+            case option
+            when :shake, :anonymous
+              option.to_s
+            when :video_only_wifi
+              'video-only-wifi'
+            else
+              UI.user_error!("Unknown option: #{option}")
+            end
+          end
+        end
+
+        client_options = Hash[params.values.map do |key, value|
+          case key
+          when :api_key
```

Comment:
Right now they have the same order as the configuration items. So it's easier not to miss any.